ocaml/tests/test_vm_check_operation_error.ml

```diff
@@ -50,6 +51,108 @@ let all_vm_operations =
   ; `update_allowed_operations
   ]
 
+let vm_op_to_string = function
+  | `assert_operation_valid ->
+      "assert_operation_valid"
+  | `awaiting_memory_live ->
+      "awaiting_memory_live"
+  | `call_plugin ->
+      "call_plugin"
+  | `changing_VCPUs ->
+      "changing_VCPUs"
+  | `changing_VCPUs_live ->
+      "changing_VCPUs_live"
+  | `changing_dynamic_range ->
+      "changing_dynamic_range"
+  | `changing_memory_limits ->
+      "changing_memory_limits"
+  | `changing_memory_live ->
+      "changing_memory_live"
+  | `changing_shadow_memory ->
+      "changing_shadow_memory"
+  | `changing_shadow_memory_live ->
+      "changing_shadow_memory_live"
+  | `changing_static_range ->
+      "changing_static_range"
+  | `changing_NVRAM ->
+      "changing_NVRAM"
+  | `checkpoint ->
+      "checkpoint"
+  | `clean_reboot ->
+      "clean_reboot"
+  | `clean_shutdown ->
+      "clean_shutdown"
+  | `clone ->
+      "clone"
+  | `copy ->
+      "copy"
+  | `create_template ->
+      "create_template"
+  | `create_vtpm ->
+      "create_vtpm"
+  | `csvm ->
+      "csvm"
+  | `data_source_op ->
+      "data_source_op"
+  | `destroy ->
+      "destroy"
+  | `export ->
+      "export"
+  | `get_boot_record ->
+      "get_boot_record"
+  | `hard_reboot ->
+      "hard_reboot"
+  | `hard_shutdown ->
+      "hard_shutdown"
+  | `import ->
+      "import"
+  | `make_into_template ->
+      "make_into_template"
+  | `metadata_export ->
+      "metadata_export"
+  | `migrate_send ->
+      "migrate_send"
+  | `pause ->
+      "pause"
+  | `pool_migrate ->
+      "pool_migrate"
+  | `power_state_reset ->
+      "power_state_reset"
+  | `provision ->
+      "provision"
+  | `query_services ->
+      "query_services"
+  | `resume ->
+      "resume"
+  | `resume_on ->
+      "resume_on"
+  | `revert ->
+      "revert"
+  | `reverting ->
+      "reverting"
+  | `send_sysrq ->
+      "send_sysrq"
+  | `send_trigger ->
+      "send_trigger"
+  | `shutdown ->
+      "shutdown"
+  | `snapshot ->
+      "snapshot"
+  | `snapshot_with_quiesce ->
+      "snapshot_with_quiesce"
+  | `start ->
+      "start"
+  | `start_on ->
+      "start_on"
+  | `suspend ->
+      "suspend"
+  | `unpause ->
+      "unpause"
+  | `update_allowed_operations ->
+      "update_allowed_operations"
```

Comment:
Could this be derived by ppx since it only  ``` `type``` to "type"?

Reply:
Then perhaps the autogenerator can create a to_string function itself?

Reply:
It should also be possible to create the _all binding for enum, instead of having to implement it